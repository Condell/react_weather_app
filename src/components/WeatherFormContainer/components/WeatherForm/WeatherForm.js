import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";

import Button from "./components/Button/Button";
import Input from "./components/Input/Input";

const Form = styled.div`
  display: flex;
  flex-flow: ${props => (props.row ? "row" : "column")};
  align-items: center;
  justify-content: space-around;
  flex: none;
`;

Form.propTypes = {
  onChange: PropTypes.func,
  children: PropTypes.array
};

const WeatherForm = props =>
  <Form {...props}>
    <Input
      type="text"
      size={15}
      placeholder="St. George, Utah"
      value={props.text}
      onChange={props.onChange}
    />

    <Button
      to={{
        pathname: "/forecast",
        search: `city=${props.text}`
      }}
    >
      Get Weather
    </Button>
  </Form>;

export default WeatherForm;
