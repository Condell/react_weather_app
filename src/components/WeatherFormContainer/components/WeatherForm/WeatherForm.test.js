import React from "react";
import { shallow } from "enzyme";
import 'jest-styled-components'
import toJson from "enzyme-to-json"

import WeatherForm from "./WeatherForm";

jest.mock("./WeatherForm.js", () => "WeatherForm")

describe("WeatherForm", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<WeatherForm />);
  });

  it("should render", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
