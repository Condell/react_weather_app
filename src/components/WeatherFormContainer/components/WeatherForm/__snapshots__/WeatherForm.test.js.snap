// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherForm should render 1`] = `
<styled.div>
  <styled.input
    placeholder="St. George, Utah"
    size={15}
    type="text"
  />
  <Styled(Link)
    to={
      Object {
        "pathname": "/forecast",
        "search": "city=undefined",
      }
    }
  >
    Get Weather
  </Styled(Link)>
</styled.div>
`;
